src/Soil.jl

```diff
@@ -0,0 +1,69 @@
+module Soil
+using ClimaLSM
+using ClimaCore
+import ClimaCore: Fields, Operators, Geometry
+using ClimaLSM.Domains
+import ClimaLSM.Domains: make_function_space, center_coordinates
+import ClimaLSM:
+    AbstractModel,
+    initialize_states,
+    initialize_auxiliary,
+    initialize_prognostic,
+    make_update_aux,
+    make_rhs,
+    make_ode_function,
+    prognostic_vars,
+    auxiliary_vars
+export RichardsModel
+
+"""
+    RichardsModel
+
+A standin for a model used to simulate water flow in soil via Richards equation.
+"""
+struct RichardsModel{FT, ps, d} <: AbstractModel{FT}
+    param_set::ps
+    domain::d
+    model_name::Symbol
+end
+
+function RichardsModel{FT}(; param_set = param_set, domain = domain) where {FT}
+    return RichardsModel{FT, typeof(param_set), typeof(domain)}(
+        param_set,
+        domain,
+        :soil,
+    )
+end
+
+prognostic_vars(model::RichardsModel) = (:ϑ_l,)
+
+auxiliary_vars(model::RichardsModel) = (:z,)
+
+function initialize_states(model::RichardsModel{FT}) where {FT}
```

Comment:
I kept this "initialize_states" wrapper for now. The issue of only creating the space once is somewhat limiting. What we had sketched out today for just the soil model worked (first get the coordinates, then initialize prognostic by passing in coordinates and `model` to some init_prognostic(model, coords), and same for aux), but it doesnt nicely generalize to an LSM with multiple components. (get the coordinates for all components? then pass all to init_prognostic(land, all_coordinates) which calls init_prognostic(soil, coordinates_soil), etc?).

Reply:
https://github.com/CliMA/ClimaLSM.jl/issues/3